src/ghostfragment/fragmenting/inter_finder.cpp

```diff
@@ -0,0 +1,76 @@
+#include "inter_finder.hpp"
+#include <iostream>
+
+using namespace ghostfragment::fragmenting;
+
+void ghostfragment::fragmenting::find_group_intersections(
+  const system_type& fragments, inter_type inter, std::size_t recent_add,
+  res_type& m, log_type& log, group_type& group) {
+    if(group.size() > 2) {
+        for(std::size_t i = 0; i < group.size() - 1; ++i) {
+            if(log.count(std::make_pair(group[i], recent_add)) == 0) return;
+        }
+        inter_type::iterator it = set_intersection(
+          inter.begin(), inter.end(), fragments[recent_add].begin(),
+          fragments[recent_add].end(), inter.begin());
+        inter.resize(it - inter.begin());
+    } else if(group.size() == 2)
+        inter = log[std::make_pair(group[0], group[1])];
+
+    if(!(inter.empty())) {
+        if(m.count(inter) > 0)
+            m[inter] += (group.size() % 2 == 0) ? -1 : 1;
+        else if(group.size() > 1)
+            m[inter] = (group.size() % 2 == 0) ? -1 : 1;
+
+        for(std::size_t new_frag = recent_add + 1; new_frag < fragments.size();
+            ++new_frag) {
+            group_type new_group(group);
+            new_group.push_back(new_frag);
+            find_group_intersections(fragments, inter, new_frag, m, log,
+                                     new_group);
+        }
+    }
+}
+
+void ghostfragment::fragmenting::find_pair_intersections(
+  const system_type& fragments, log_type& log) {
+    inter_type::iterator it;
+    for(std::size_t i = 0; i < fragments.size(); ++i) {
+        for(std::size_t j = i + 1; j < fragments.size(); ++j) {
+            inter_type inter(20);
```

Comment:
Magic numbers like this are bad ideas. For example, this one will lead to undefined behavior if the intersection has more than 20 atoms in it. Since, the intersection can't possibly be bigger than either fragment, you can default the size to one of the two fragments (the minimum size ideally).

Reply:
This is just a small hold-back from the initial problem. I was under the impression that 20 atoms would be the max size of _a_ fragment, but that isn't necessarily the case. I didn't even think of the possibility of using the actual fragment size.

Reply:
For all practical purposes I don't think there's a max size. 20 atoms was my guess for an average size.